### ISSUE TITLE
[BUG] fanrpm not shown in HomeAssistant

### ISSUE THREAD

**Issue by andreas-bulling** [revk/ESP32-Faikin#371]
I just noticed that the fanrpm are not properly sent to HomeAssistant. It's shown as "unknown".

![image](https://github.com/revk/ESP32-Faikin/assets/377856/ce69e896-1d75-4d69-8cee-0a6915ab2504)

Firmware: 4cc882f

**Comment by revk** [comment]
Can you confirm what you see in the MQTT.

**Comment by andreas-bulling** [comment]
The AC was set to cooling mode

![image](https://github.com/revk/ESP32-Faikin/assets/377856/145ddf0e-e77c-446a-b947-40880220e665)

**Comment by revk** [comment]
Rather odd if seeing `fanrpm` as `0` and as an array. The array is typical for the normal `.../status` messages the Faikin sends on MQTT every minute. Also HA is looking for `fanfreq`.

Check the topic HA is looking for and post what is in that topic?

**Comment by andreas-bulling** [comment]
I should add: Faiking didn't show up at first after opening MQTT Explorer. Only once I turned one of the ACs on did these values appear.

`fanfreq` is also there (didn't screengrab it).

![image](https://github.com/revk/ESP32-Faikin/assets/377856/c20b76e7-5b50-4ec7-913e-eb28bf969124)


I am afraid I don't know what you mean/what else I should check. I mean - I configured MQTT for Faikin at some point, it showed up in the MQTT integration in HA, and HA automatically converted all values into corresponding entities... So I assume HA is looking for the topics it has converted itself?!

**Comment by andreas-bulling** [comment]
If I may add - and maybe you already have that: But given that you seem to have quite a number of HA users, wouldn't it make sense for you to setup a small HA system to be able to look into/debug these kind of things yourself?

Just because there seems to be quite a bit of time-consuming back and forth as users have to provide information, don't deliver the right or incomplete information, you have to ask again, there can be misunderstandings...

HA is not at all difficult to setup on a small/old RPi for example. Or even in a virtual machine.

I think this would speed up things in a few cases.

**Comment by revk** [comment]
I do plan a test HA set up at some point, yet. I've been busy.

But there are so many models, my tests on my units will not get to the bottom of most issues anyway.

**Comment by revk** [comment]
OK, Home Assistant makes this easy.

Firstly, it works.

<img width="342" alt="Screenshot 2024-06-28 at 11 49 54" src="https://github.com/revk/ESP32-Faikin/assets/996983/a0a7ede1-036c-49f0-b406-86372686fa63">

But in mqtt, selecting the device, you can see MQTT info, which advises what it is looking for.

<img width="595" alt="Screenshot 2024-06-28 at 11 50 39" src="https://github.com/revk/ESP32-Faikin/assets/996983/c445099c-d913-4ded-9de7-5580beee9e0a">

And it even shows the messages it has received...

<img width="253" alt="Screenshot 2024-06-28 at 11 50 52" src="https://github.com/revk/ESP32-Faikin/assets/996983/f257c414-f79c-4ab0-bbad-aa5151cb3324">

So, maybe you can check those details for yours...

**Comment by andreas-bulling** [comment]
Nice - but where do I need to go in HA to be "in mqtt" ?

**Comment by revk** [comment]
Err, lets look!!!

<img width="663" alt="Screenshot 2024-06-28 at 14 21 21" src="https://github.com/revk/ESP32-Faikin/assets/996983/16755b9f-9878-4e5a-bf53-0387ce56adb7">
<img width="401" alt="Screenshot 2024-06-28 at 14 21 26" src="https://github.com/revk/ESP32-Faikin/assets/996983/db9c5b91-779b-4d34-a876-b0acddcc0410">
<img width="1094" alt="Screenshot 2024-06-28 at 14 21 32" src="https://github.com/revk/ESP32-Faikin/assets/996983/aeda4b3a-7064-4e1b-a214-88610b22b941">
<img width="1083" alt="Screenshot 2024-06-28 at 14 21 51" src="https://github.com/revk/ESP32-Faikin/assets/996983/b2fbd2e7-5084-4c6e-9170-d87c527345a7">
<img width="372" alt="Screenshot 2024-06-28 at 14 21 56" src="https://github.com/revk/ESP32-Faikin/assets/996983/83e9b9ad-f80a-464a-90f2-b9a72fb17a31">
<img width="504" alt="Screenshot 2024-06-28 at 14 22 03" src="https://github.com/revk/ESP32-Faikin/assets/996983/4c317314-af48-48d1-a1d9-3722011801c4">

**Comment by andreas-bulling** [comment]
I've been using HA for years now and have never clicked on these three dots - thanks a lot!

**Comment by Sonic-Amiga** [comment]
One of my users has seen this too... I see you have "fanfreq"; and at some point fanrpm was superseded by fanfreq. Apparently HA caches old data items internally. Consider that a HA feature. I am not using HA myself, so can't give any advice, sorry.

BTW, the newest Faikin allows the user to choose between rpm and frequency (in Hz) in advanced settings.

**Comment by revk** [comment]
It could be cached in the MQTT server as a retained message - restarting MQTT may clear.
